박민영/Math (수학)/백준/25206.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * @author Minyeong Park
+ * @date 2023.03.05.
+ * 너의 평점은
+ * 문제 링크 : https://www.acmicpc.net/problem/25206
+ */
+
+
+
+import java.io.*;
+import java.util.StringTokenizer;
+
+public class Main {
+    public static void main(String[] args) throws IOException {
+        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+        String line;
+        StringTokenizer st;
+        double grade_sum = 0.0;
+        double size_sum = 0.0;
+        for (int i = 0; i < 20; i++) {
+            st = new StringTokenizer(br.readLine());
+            st.nextToken();
+            double size = Double.parseDouble(st.nextToken());
+            double grade = 0.0;
+            String grade_str = st.nextToken();
+            switch (grade_str) {
+                case "A+":
+                    grade = 4.5;
+                    break;
+                case "A0":
+                    grade = 4.0;
+                    break;
+                case "B+":
+                    grade = 3.5;
+                    break;
+                case "B0":
+                    grade = 3.0;
+                    break;
+                case "C+":
+                    grade = 2.5;
+                    break;
+                case "C0":
+                    grade = 2.0;
+                    break;
+                case "D+":
+                    grade = 1.5;
+                    break;
+                case "D0":
+                    grade = 1.0;
+                    break;
+                case "F":
+                    grade = 0.0;
+                    break;
+                case "P":
+                    break;
+            }
+
+            if (!grade_str.equals("P")) {
+                grade_sum += size * grade;
+                size_sum += size;
+            }
```

Comment:
```suggestion
                case "P":
                    continue;
            }

            grade_sum += size * grade;
            size_sum += size;
```

이렇게 코드를 조금 더 간략하게 쓸 수 있을 것 같네요!

Reply:
헐 우와 제가 푼 코드 리팩토링까지.. 헉 정말 감사해요!! 
오옷 "P"일 때에도 어차피 성적은 0으로 들어가니까 똑같이 돌려도 되는군요! 우와 진짜 많이 배워갑니다.. 채연이 최고:smile::+1:

Reply:
앗 잘못 이해했네요..  `continue;`로 써서 아예 다음 for문으로 진행하게 하는 것이였군요! 헐 대박 당신은 천재.. 감사합니다!!:heart: